module/os/linux/spl/spl-proc.c

```diff
@@ -668,6 +672,16 @@ static void spl_proc_cleanup(void)
 	remove_proc_entry("taskq", proc_spl);
 	remove_proc_entry("spl", NULL);
 
+#ifndef HAVE_REGISTER_SYSCTL_TABLE
+	if (spl_kstat) {
+		unregister_sysctl_table(spl_kstat);
+		spl_kstat = NULL;
+	}
+	if (spl_kmem) {
+		unregister_sysctl_table(spl_kmem);
+		spl_kmem = NULL;
+	}
```

Comment:
Just reading the patch, I've one question: shouldn't this "#ifndef HAVE_REGISTER_SYSCTL_TABLE" be extended to the third unregister_sysctl_table call just below? Same in line 50 BTW: why is spl_header not included in the #ifndef?

Reply:
@ceedriic the same `spl_header` pointer is used both cases (`HAVE_REGISTER_SYSCTL_TABLE` and `!HAVE_REGISTER_SYSCTL_TABLE`), that's why it's outside the `#ifndef` blocks.